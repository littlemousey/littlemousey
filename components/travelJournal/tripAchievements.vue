<template>
  <v-layout row wrap align-start justify-center>
    <v-flex xs12 sm10 md4 my-5>
      <v-card>
        <v-toolbar color="blue darken-4" dark>
          <v-toolbar-title>Achievements üê≠</v-toolbar-title>
        </v-toolbar>
        <v-list>
          <v-list-tile
            v-for="achievement in achievements"
            :key="achievement.title"
            avatar
          >
            <v-list-tile-action>
              <v-icon v-if="achievement.achieved" color="red">star</v-icon>
            </v-list-tile-action>

            <v-list-tile-content>
              <v-list-tile-title v-text="achievement.title"></v-list-tile-title>
            </v-list-tile-content>
          </v-list-tile>
        </v-list>
      </v-card>
    </v-flex>
  </v-layout>
</template>

<script>
export default {
  name: 'TripAchievements',
  props: {
    achievements: {
      type: Array,
      required: true
    }
  }
}
</script>

<style lang="scss" scoped>
.v-list__tile__title {
  font-size: 12px;
}
</style>
