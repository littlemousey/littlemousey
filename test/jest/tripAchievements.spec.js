import { shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import tripAchievements from '@/components/travelJournal/tripAchievements'

Vue.use(Vuetify)

const wrapper = shallowMount(tripAchievements, {
  propsData: {
    title: 'title',
    achievements: [{ title: 'title', achieved: true }]
  }
})

describe('Achievements', () => {
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
