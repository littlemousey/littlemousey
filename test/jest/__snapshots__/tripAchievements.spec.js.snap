// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Achievements renders correctly 1`] = `
<v-row-stub
  align="start"
  justify="center"
  tag="div"
>
  <v-col-stub
    class="my-5"
    cols="12"
    md="4"
    sm="10"
    tag="div"
  >
    <v-card-stub
      loaderheight="4"
      tag="div"
    >
      <v-toolbar-stub
        color="blue darken-4"
        dark="true"
        extensionheight="48"
        src=""
        tag="header"
        tile="true"
      >
        <v-toolbar-title-stub>
          title
        </v-toolbar-title-stub>
      </v-toolbar-stub>
       
      <v-list-stub
        tag="div"
        tile="true"
      >
        <v-list-item-stub
          activeclass=""
          tag="div"
        >
          <v-list-item-action-stub>
            <v-icon-stub
              color="red"
            >
              
            </v-icon-stub>
          </v-list-item-action-stub>
           
          <v-list-item-content-stub>
            <v-list-item-title-stub />
          </v-list-item-content-stub>
        </v-list-item-stub>
      </v-list-stub>
    </v-card-stub>
  </v-col-stub>
</v-row-stub>
`;
